src/main/java/org/opensearch/ad/task/ADTaskManager.java

```diff
@@ -520,12 +613,14 @@ public void stopDetector(
                 return;
             }
             SearchHit searchHit = r.getHits().getAt(0);
-            try (XContentParser parser = createXContentParserFromRegistry(xContentRegistry, searchHit.getSourceRef())) {
+            try (XContentParser parser = RestHandlerUtils.createXContentParserFromRegistry(xContentRegistry, searchHit.getSourceRef())) {
                 ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
                 ADTask adTask = ADTask.parse(parser, searchHit.getId());
+                logger.debug("latest task id is {}, for detector {}", adTask.getTaskId(), adTask.getDetectorId());
 
-                // TODO: support resetting realtime task as stopped
-                if (adTask.isHistoricalTask() && !isADTaskEnded(adTask) && lastUpdateTimeExpired(adTask)) {
+                // TODO: check realtime detector job and reset realtime task as stopped.
+                if (resetTaskState && adTask.isHistoricalTask() && !isADTaskEnded(adTask) && lastUpdateTimeExpired(adTask)) {
```

Comment:
Copy from https://github.com/opensearch-project/anomaly-detection/pull/58#discussion_r658370693

> when the coordinating node is gone, would the reset happen?
> When the worker node is gone, would the coordinating node wait forever?

Will reset task when get task, and have hourly cron to maintain tasks. These changes will be in another PR, will send out later.